rq/worker.py

```diff
@@ -1569,7 +1603,7 @@ class RoundRobinWorker(Worker):
 
     def reorder_queues(self, reference_queue):
```

Comment:
Can we also move `reorder_queues` from line 828 into `dequeue_job_and_maintain_ttl()`

Reply:
Sure

Reply:
I think the way the test work right now depends on when this is called, so when moving to the dequeue method, tests break. Since this is a refactor, and not related to the feature itself, my suggestion is to work on this on a new PR, we will have to chose how to handle the not having the `reference_queue` 

_(I think you mentioned doing something like `self._ordered_queues[-1:][0]` to get the last element, but this would possibly raise an `IndexError`, so we would have to change the signature to make for a safe flow._

Reply:
Ok, in this case no need to pass `dequeue_strategy` to `bootstrap()` just to be assigned to `self` here: https://github.com/rq/rq/pull/1806/files#diff-50984f276e8a6fea5f4ba4938afb6f4fbc544fd3751fbd16027dc0a326d6405eR734

Just assign `._dequeue_strategy` when `work` is called.

Reply:
Hmm, not sure I follow the reasoning here - Isn't this use case exactly why the `bootstrap` method was created for - initial tasks that are generic enough that any worker would use?